In [11]:
# importing necessary libraries for our work
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib as plt

In [149]:
# reading amazon prime dataset from csv file into our data variable
data = pd.read_csv("amazon_prime_titles.csv")

In [16]:
# printing last 5 rows of the dataset
data.tail(5)

show_id     type                    title         director  \
9663   s9664    Movie      Pride Of The Bowery  Joseph H. Lewis   
9664   s9665  TV Show            Planet Patrol              NaN   
9665   s9666    Movie                  Outpost     Steve Barker   
9666   s9667  TV Show  Maradona: Blessed Dream              NaN   
9667   s9668    Movie              Harry Brown    Daniel Barber   

                                                   cast country date_added  \
9663                           Leo Gorcey, Bobby Jordan     NaN        NaN   
9664  DICK VOSBURGH, RONNIE STEVENS, LIBBY MORRIS, M...     NaN        NaN   
9665  Ray Stevenson, Julian Wadham, Richard Brake, M...     NaN        NaN   
9666  Esteban Recagno, Ezequiel Stremiz, Luciano Vit...     NaN        NaN   
9667  Michael Caine, Emily Mortimer, Joseph Gilgun, ...     NaN        NaN   

      release_year rating   duration                listed_in  \
9663          1940     7+     60 min                   Comedy   
9664          2018    13+  4 Seasons                 TV Shows   
9665          2008      R     90 min                   Action   
9666          2021  TV-MA   1 Season            Drama, Sports   
9667          2010      R    103 min  Action, Drama, Suspense   

                                            description  
9663  New York City street principles get an East Si...  
9664  This is Earth, 2100AD - and these are the adve...  
9665  In war-torn Eastern Europe, a world-weary grou...  
9666  The series tells the story of Diego Maradona, ...  
9667  Harry Brown, starring two-time Academy Award w...

In [8]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9668 entries, 0 to 9667
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       9668 non-null   object
 1   type          9668 non-null   object
 2   title         9668 non-null   object
 3   director      7586 non-null   object
 4   cast          8435 non-null   object
 5   country       672 non-null    object
 6   date_added    155 non-null    object
 7   release_year  9668 non-null   int64 
 8   rating        9331 non-null   object
 9   duration      9668 non-null   object
 10  listed_in     9668 non-null   object
 11  description   9668 non-null   object
dtypes: int64(1), object(11)
memory usage: 906.5+ KB


In [99]:
list(data.columns)

['show_id',
 'type',
 'title',
 'director',
 'cast',
 'country',
 'date_added',
 'release_year',
 'rating',
 'duration',
 'listed_in',
 'description']

**Info**: There are 9668 rows and 12 columns in the dataframe.
*   Respectively, these columns are:
    *   show_id,
    *   type,   
    *   title,
    *   director,
    *   cast,
    *   country,
    *   date_added,
    *   release_year,
    *   rating,
    *   duration,
    *   listed_in,
    *   description

In [24]:
data['rating'].describe()

count     9331
unique      24
top        13+
freq      2117
Name: rating, dtype: object

There are 9331 rows with filled rating value, 24 unique type of rating. Most occurring rating type is 13+ with a frequency of 2117 rows.

In [35]:
data['rating'].value_counts()

13+         2117
16+         1547
ALL         1268
18+         1243
R           1010
PG-13        393
7+           385
PG           253
NR           223
TV-14        208
TV-PG        169
TV-NR        105
G             93
TV-G          81
TV-MA         77
TV-Y          74
TV-Y7         39
UNRATED       33
NC-17          3
AGES_18_       3
NOT_RATE       3
AGES_16_       2
16             1
ALL_AGES       1
Name: rating, dtype: int64

In [36]:
data['type'].value_counts()

Movie      7814
TV Show    1854
Name: type, dtype: int64

There are 7814 Movies and 1854 TV Shows inside the dataset

In [43]:
min_release = data['release_year'].min()
#min(data['release_year'])
print(min_release)
max_release = data['release_year'].max()
print(max_release)

1920
2021


In [64]:
data[data.release_year == min_release].count()

show_id         3
type            3
title           3
director        3
cast            3
country         0
date_added      0
release_year    3
rating          3
duration        3
listed_in       3
description     3
dtype: int64

In [61]:
data[data.release_year == max_release].count()

show_id         1442
type            1442
title           1442
director        1123
cast            1232
country           59
date_added        53
release_year    1442
rating          1425
duration        1442
listed_in       1442
description     1442
dtype: int64

There are 3 entries with the oldest release year(1920) and 1442 entries with the newest release year(2021)

In [108]:
# checking if there are any columns with null value
data.isnull().any()

show_id         False
type            False
title           False
director         True
cast             True
country          True
date_added       True
release_year    False
rating           True
duration        False
listed_in       False
description     False
dtype: bool

In [110]:
data.isnull().sum()

show_id            0
type               0
title              0
director        2082
cast            1233
country         8996
date_added      9513
release_year       0
rating           337
duration           0
listed_in          0
description        0
dtype: int64

In [132]:
# filtering columns with more than 500 null values
na_cols = list(data.columns[data.isnull().sum() > 500])
na_cols

['director', 'cast', 'country', 'date_added']

In [154]:
# filling null values with empty strings in columns with more than 500 null values
data[na_cols] = data[na_cols].fillna('')

In [153]:
data.isnull().sum()

show_id           0
type              0
title             0
director          0
cast              0
country           0
date_added        0
release_year      0
rating          337
duration          0
listed_in         0
description       0
dtype: int64

In [171]:
# dropping the entries with the null values in rating column
data.dropna(inplace=True)
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 9331 entries, 1 to 9667
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       9331 non-null   object
 1   type          9331 non-null   object
 2   title         9331 non-null   object
 3   director      9331 non-null   object
 4   cast          9331 non-null   object
 5   country       9331 non-null   object
 6   date_added    9331 non-null   object
 7   release_year  9331 non-null   int64 
 8   rating        9331 non-null   object
 9   duration      9331 non-null   object
 10  listed_in     9331 non-null   object
 11  description   9331 non-null   object
dtypes: int64(1), object(11)
memory usage: 947.7+ KB
